From 06d6f67889df9ce23725dba6e1cdb06cf3ab7999 Mon Sep 17 00:00:00 2001
From: whrvt <william@horvath.blog>
Date: Wed, 17 Jan 2024 04:07:05 -0800
Subject: [PATCH 02/41] Revert "server: Introduce a new
 get_desktop_cursor_thread_input helper."

This reverts commit 9dc80b13bec0b482150dbd8c9236dbb8b6a31ccd.

 100.0% server/
diff --git a/server/queue.c b/server/queue.c
index d2cf9800258..ee8bdf34b7f 100644
--- a/server/queue.c
+++ b/server/queue.c
@@ -429,34 +429,26 @@ static void queue_cursor_message( struct desktop *desktop, user_handle_t win, un
     queue_hardware_message( desktop, msg, 1 );
 }
 
-static struct thread_input *get_desktop_cursor_thread_input( struct desktop *desktop )
-{
-    struct thread_input *input = NULL;
-    struct thread *thread;
-
-    if ((thread = get_window_thread( desktop->cursor.win )))
-    {
-        if (thread->queue) input = thread->queue->input;
-        release_object( thread );
-    }
-
-    return input;
-}
-
 static int update_desktop_cursor_window( struct desktop *desktop, user_handle_t win )
 {
     int updated = win != desktop->cursor.win;
-    struct thread_input *input;
+    user_handle_t handle = desktop->cursor.handle;
     desktop->cursor.win = win;
-
-    if (updated && (input = get_desktop_cursor_thread_input( desktop )))
+    if (updated)
     {
-        user_handle_t handle = input->cursor_count < 0 ? 0 : input->cursor;
+        struct thread *thread;
+
+        if ((thread = get_window_thread( win )))
+        {
+            struct thread_input *input = thread->queue->input;
+            if (input) handle = input->cursor_count < 0 ? 0 : input->cursor;
+            release_object( thread );
+        }
+
         /* when clipping send the message to the foreground window as well, as some driver have an artificial overlay window */
         if (is_cursor_clipped( desktop )) queue_cursor_message( desktop, 0, WM_WINE_SETCURSOR, win, handle );
         queue_cursor_message( desktop, win, WM_WINE_SETCURSOR, win, handle );
     }
-
     return updated;
 }
 
-- 
2.43.0

